src/app/(auth)/subscriber/page.tsx

```diff
@@ -1,39 +1,51 @@
 'use client';
 
 import Button from '@/components/Button';
-import { getSubscribers } from '@/apis/subscriber';
-import Input from '@/components/Input';
+import { getSubscribers, deleteSubscriber } from '@/apis/subscriber';
 import Table from '@/components/Table';
 import { useEffect, useState } from 'react';
-import { Subscribers, TableData } from '@/utils/types';
+import { SubscriberResource, TableData } from '@/utils/types';
 import { UserGroupIcon } from '@heroicons/react/24/outline';
 import { PencilSquareIcon, TrashIcon } from '@heroicons/react/24/solid';
+import Link from 'next/link';
+import { toast } from 'react-toastify';
 
 const Page = () => {
-  const [subscribers, setSubscribers] = useState<Subscribers | undefined>(
-    undefined
-  );
+  const [subscribers, setSubscribers] = useState<
+    SubscriberResource | undefined
+  >(undefined);
 
-  useEffect(() => {
-    const fetchSubscribers = async () => {
-      setSubscribers(await getSubscribers());
-    };
-    fetchSubscribers();
-  }, []);
+  const fetchSubscribers = async () => {
+    setSubscribers(await getSubscribers());
+  };
 
   const [tableData, setTableData] = useState<TableData>({
     headers: ['Details', 'List'],
     rows: [],
   });
 
   useEffect(() => {
+    fetchSubscribers();
+    const showDeleteConfirmationModal = async (id: number) => {
```

Comment:
why this function defined inside `useEffect`? Any benefit of it?

Reply:
Yes, Earlier if I was using this function outside the use effect then getting some error. So after researching by defining this function inside the use effect the component logic and handle the asynchronous operation are grouped together and also it's making easier to work with promises and API calls. And also I think it's can improve performance.